mvpa2/clfs/gnb.py

```diff
@@ -289,36 +299,58 @@ def _predict(self, data):
 
         return predictions
 
+    def get_sensitivity_analyzer(self, **kwargs):
+        """Returns a sensitivity analyzer for GNB if GNB is linear (i.e. if common_variance=True)."""
+        params = self.params
+        if params.common_variance: # == True:
+            return GNBWeights(self, **kwargs)
+        else:
+            raise NotImplementedError("Sensitivity calculation is only sensible for a linear GNB, which is true when "
+                            "common_variances = True. Did you forget to specify this parameter?")
 
-    # XXX Later come up with some
-    #     could be a simple t-test maps using distributions
-    #     per each class
-    #def get_sensitivity_analyzer(self, **kwargs):
-    #    """Returns a sensitivity analyzer for GNB."""
-    #    return GNBWeights(self, **kwargs)
-
-
-    # XXX Is there any reason to use properties?
-    #means = property(lambda self: self.__biases)
-    #variances = property(lambda self: self.__weights)
-
-
-
-## class GNBWeights(Sensitivity):
-##     """`SensitivityAnalyzer` that reports the weights GNB trained
-##     on a given `Dataset`.
-
-##     """
-
-##     _LEGAL_CLFS = [ GNB ]
 
-##     def _call(self, dataset=None):
-##         """Extract weights from GNB classifier.
+class GNBWeights(Sensitivity):
+    """
+    `SensitivityAnalyzer` that reports the weights GNB trained
+    on a given `Dataset`.
+    """
 
-##         GNB always has weights available, so nothing has to be computed here.
-##         """
-##         clf = self.clf
-##         means = clf.means
-##           XXX we can do something better ;)
-##         return means
+    _LEGAL_CLFS = [ GNB ]
+
+    def _call(self, dataset):
+    # compute weights as difference between observed value and attribute mean given class
+    # for decision in all pairwise combinations of labels
+        import itertools
+        clf=self.clf
+        # get means of all attributes given class label
+        means = clf.means
+        # number of features
+        nfeat = clf.means.shape[1]
+
+        # make sure labels are numeric
+        am = AttributeMap()
+        ulab = am.to_numeric(clf.ulabels)
+        # all pairwise combinations of labels
+        lcomb = list(itertools.combinations(ulab, 2))
+
+        weights = np.zeros([len(lcomb), nfeat])
+        # do not compute sensitivity for features with variance 0 as this would implicate
+        # a division by zero
+        zero_vars = clf.variances==0
+        for pair in lcomb:
+            for i in range(0, nfeat):
```

Comment:
just rely on numpy's ability to work with vectors. So here you really want just
`python
weights[row_idx] = (means[pair[0]] - means[pair[1])/stddev
`